src/components/diff/DiffWidget.tsx

```diff
@@ -0,0 +1,98 @@
+import { Widget } from '@phosphor/widgets';
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import { IRenderMimeRegistry } from '@jupyterlab/rendermime';
+
+import { getRefValue, IDiffContext } from './model';
+import { Diff, isDiffSupported } from './Diff';
+import { JupyterLab } from '@jupyterlab/application';
+import { showDialog } from '@jupyterlab/apputils';
+import { PathExt } from '@jupyterlab/coreutils';
+import { style } from 'typestyle';
+
+export class DiffWidget extends Widget {
```

Comment:
I believe this should extend `ReactWidget` instead of `Widget`. Then you fill in the `render` method and you don't have to mess with `ReactDOM.render`...

Reply:
This PR is on pre-1.0 changes and since nbdime isn't on 1.0 yet, and AFAIK `ReactWidget` isn't available on pre-1.0. I plan to refactor to use `ReactWidget` once